src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java

```diff
@@ -163,7 +159,10 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   @VisibleForTesting
   /** Returns YouTube video ID for a given {@code videoUrl}. */
-  protected Optional<String> getVideoId(String videoUrl) {
+  // TODO: Move this function to a Utils class.
+  public static Optional<String> getVideoId(String videoUrl) {
+    /* Pattern used to create a matcher for a video ID. */
+    Pattern videoUrlGeneratedPattern = Pattern.compile(YOUTUBE_VIDEO_URL_PATTERN);
```

Comment:
Why is this no longer static? Keep the compiled pattern static means we don't have to recompile it every time it's used